anndata/readwrite/read.py

```diff
@@ -197,8 +199,15 @@ def read_text(
         Numpy data type.
     """
     if isinstance(filename, (str, Path)):
-        with Path(filename).open() as f:
-            return _read_text(f, delimiter, first_column_names, dtype)
+        if str(filename).endswith('.gz'):
+            with gzip.open(str(filename), mode='rt') as f:
```

Comment:
This is also possible without str() but requires Python >= 3.6, so I wasn't sure. https://bugs.python.org/issue28227